src/NovelRT.Interop/Input/NrtInputAction.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#include <NovelRT.Interop/Input/NrtInputAction.h>
+#include <NovelRT.Interop/Input/NrtNovelKey.h>
+#include <NovelRT.Interop/NrtErrorHandling.h>
+#include <NovelRT/Input/Input.h>
+
+using namespace NovelRT::Input;
+
+NrtInputActionInfo Nrt_InputAction_ReadAsInfo(NrtInputActionHandle action)
+{
+    auto* inputAction = reinterpret_cast<InputAction*>(action);
+    return NrtInputActionInfo{inputAction->actionName.c_str(),
+                              Nrt_NovelKey_ReadAsInfo(reinterpret_cast<NrtNovelKeyHandle>(&inputAction->pairedKey)),
+                              static_cast<NrtKeyState>(inputAction->state)};
+}
+intptr_t Nrt_InputActionList_IncrementSize()
+{
+    return sizeof(InputAction);
+}
```

Comment:
If this were to be a `constexpr` function in C++, I believe it would become compile-time and therefore work as a constant (since `sizeof` is considered compile-time as well), no? Without that I agree - it'd be a runtime function and wouldn't provide the same benefit as expected.

Reply:
This is C

constexpr does not exist

:^)

Reply:
Also P/Invoke will not care for your constexpr :D

Reply:
Well its getting exposed as C, but its a c++ function 😛 (otherwise this file should end in `.c`)
Anyways if C# isn't going to care for it (how would Java then???), then either way it does appear to be moot

Reply:
You can't make a C compliant function constexpr even if it is internally C++

Reply:
> I think C# would need to define its own constant. This function is at runtime. I don't entirely follow how this would work.

That's the problem. C# *can't* guess the size of `InputAction`. This is why it can make it a sort-of constant:
```cs
public static readonly nuint Size = Nrt_InputActionList_IncrementSize();
```

Reply:
Gotcha. Thanks. :D